server/main.lua

```diff
@@ -5,95 +5,133 @@
+----   Functions   ----
 -----------------------
 
--- Event to give keys. receiver can either be a single id, or a table of ids.
--- Must already have keys to the vehicle, trigger the event from the server, or pass forcegive paramter as true.
-RegisterNetEvent('qb-vehiclekeys:server:GiveVehicleKeys', function(receiver, plate)
-    local giver = source
+local function getCitizenId(source)
+    local player = exports.qbx_core:GetPlayer(source)
+    if not player then return end
 
-    if HasKeys(giver, plate) then
-        exports.qbx_core:Notify(giver, locale("notify.gave_keys"))
-        if type(receiver) == 'table' then
-            for r in receiver do
-                GiveKeys(receiver[r], plate)
-            end
-        else
-            GiveKeys(receiver, plate)
+    return player.PlayerData.citizenid
+end
+
+local function findVehicleByPlate(plate)
+    local vehicles = GetAllVehicles()
+    for i = 1, #vehicles do
+        local vehicle = vehicles[i]
+        if qbx.getVehiclePlate(vehicle) == plate then
+            return vehicle
         end
-    else
-        exports.qbx_core:Notify(giver, locale("notify.no_keys"))
     end
-end)
-
-RegisterNetEvent('qb-vehiclekeys:server:AcquireVehicleKeys', function(plate)
-    GiveKeys(source, plate)
-end)
-
-RegisterNetEvent('qb-vehiclekeys:server:breakLockpick', function(itemName)
-    if not (itemName == "lockpick" or itemName == "advancedlockpick") then return end
-    exports.ox_inventory:RemoveItem(source, itemName, 1)
-end)
+end
 
-RegisterNetEvent('qb-vehiclekeys:server:setVehLockState', function(vehNetId, state)
-    SetVehicleDoorsLocked(NetworkGetEntityFromNetworkId(vehNetId), state)
-end)
+---Loads a players vehicles to the vehicleList
+---@param src integer
+local function addPlayer(src)
+    local citizenid = getCitizenId(src)
+    if not citizenid then return end
 
-local function getCitizenId(source)
-    local player = exports.qbx_core:GetPlayer(source)
+    if not keysList[citizenid] then
+        keysList[citizenid] = {}
+    end
 
-    if not player then return end
-    local citizenid = player.PlayerData.citizenid
+    local vehicles = MySQL.query.await('SELECT * FROM player_vehicles WHERE citizenid = ?', { citizenid })
+    for i = 1, #vehicles do
+        local data = vehicles[i]
+        if findVehicleByPlate(data.plate) then
+            keysList[citizenid][data.plate] = true
+        end
+    end
 
-    return citizenid
+    Player(src).state:set('keysList', keysList[citizenid], true)
 end
 
-RegisterNetEvent('qbx-vehiclekeys:server:setPlayerKeys', function()
-    local src = source
+---Removes a players vehicles from the vehicleList
+---@param src integer
+local function removePlayer(src)
     local citizenid = getCitizenId(src)
-    Player(src).state.keysList = keysList[citizenid]
-end)
+    if not citizenid or not keysList[citizenid] then return end
 
------------------------
-----   Functions   ----
------------------------
+    keysList[citizenid] = nil
```

Comment:
There is no point in deleting this list on the user's logout.
This is the list for restoring keys after logging in again.